File: MiningActionManagementSystem/backend/app/schemas/enums.py
import enum


class ImgPath(enum.Enum):
    ACTION = "/app/app/attachments/imgs/action"


class DocumentPaths(enum.Enum):
    GENERAL = "/attachments/general"


class PriorityEnum(enum.Enum):
    UNKNOWN = 5
    LOW = 4
    MEDIUM = 3
    HIGH = 2
    URGENT = 1


class SystemTypeEnum(enum.Enum):
    ACE = "ACE"
    AHM = "AHM"
    BMS_ACT = "BMS-ACT"
    BMS_CR = "BMS-CR"
    BMS_HZD = "BMS-HZD"
    BMS_ITR = "BMS_ITR"
    SMH = "SMH"
    TEAMS = "Teams"
    SAP_NOTIFICATION = "SAP Notification"
    SAP_WORK_ORDER = "SAP Work Order"
    DEP = "DEP"


class PrivacyEnum(enum.IntEnum):
    PUBLIC = 1
    CONFIDENTIAL = 3


class LinkedSystemEnum(enum.Enum):
    ACTION = 1
    SAP_NOTIFICATION = 2
    SAP_WO = 3
    AHM_OVERDUE = 4
    AHM_SIMP = 5
    AHM_ALS = 6
    AHM_TX = 7


class ActionTagEnum(enum.Enum):
    SHORT_TERM = 1
    LONG_TERM = 2
    ONLINE = 3
    SHUTDOWN = 4
    INSPECTION_ONLY = 5
    PARTS_REQUIRED = 6
    CAPITAL = 7


class StatusEnum(enum.IntEnum):
    OVERDUE = 1
    OPEN = 2
    ON_HOLD = 3
    CLOSED = 4


class Priority(enum.IntEnum):
    OPEN = 0
    ONHOLD = 1
    CLOSED = 2


class FeedbackReason(enum.IntEnum):
    BUG = 1
    GENERAL_FEEDBACK = 2


class FeedbackStatusEnum(enum.IntEnum):
    OPEN = 1
    ON_HOLD = 2
    CLOSED_COMPLETE = 3
    CLOSED_DUPLICATE = 4
    CLOSED_FEEDBACK_ONLY = 5


class ArchiveStatus(enum.IntEnum):
    Active = 1
    Archived = 2


class LogTypeEnum(enum.Enum):
    LOGIN = "Login"
    ACTION_INGESTION_SYNC = "Action_Ingestion_Sync"
    TEAMS_INGESTION_SYNC = "Teams_Ingestion_Sync"
    ACTION_CREATION = "Action_Creation"
    ACTION_UPDATE = "Action_Update"
    WORKGROUP_CREATION = "Workgroup_Creation"
    WORKGROUP_UPDATE = "Workgroup_Update"

File: MiningActionManagementSystem/frontend/plugins/enums.js
export default ({ app }, inject) => {
  inject('enums', {
    priority: {
      1: 'Urgent',
      2: 'High',
      3: 'Medium',
      4: 'Low',
      5: 'Unknown',
    },
    source_systems: {
      ACE: 'ACE',
      AHM: 'AHM',
      BMS_ACT: 'BMS-ACT',
      BMS_CR: 'BMS-CR',
      BMS_HZD: 'BMS-HZD',
      BMS_ITR: 'BMS-ITR',
      DEP: 'DEP',
      SAP_Notification: 'SAP Notification',
      SAP_Work_Order: 'SAP Work Order',
      SMH: 'SMH',
      Teams: 'Teams',
    },
    source_logos: {
      ACE: {
        path: '/ace_logo_light.svg',
        width: 50,
      },
      AHM: {
        path: '/system_logos/ahm.png',
        width: 80,
      },
      BMS_ACT: {
        path: '/system_logos/bms.png',
        width: 60,
      },
      BMS_CR: {
        path: '/system_logos/bms.png',
        width: 60,
      },
      BMS_HZD: {
        path: '/system_logos/bms.png',
        width: 60,
      },
      BMS_ITR: {
        path: '/system_logos/bms.png',
        width: 60,
      },
      DEP: {
        path: '/system_logos/dep.png',
        width: 100,
      },
      SAP_Notification: {
        path: '/system_logos/sap.png',
        width: 60,
      },
      SAP_Work_Order: {
        path: '/system_logos/sap.png',
        width: 60,
      },
      SMH: {
        path: '/system_logos/smh.png',
        width: 60,
      },
      Teams: {
        path: '/system_logos/teams.png',
        width: 40,
      },
    },
    status: {
      1: 'Overdue',
      2: 'Open',
      3: 'On Hold',
      4: 'Closed',
    },
    archive_status: {
      1: 'Active',
      2: 'Archived',
    },
    // -------------------------------------------
    // WORKGROUP
    privacy: {
      1: 'Public',
      3: 'Confidential',
    },

    action_privacy: {
      1: 'Public',
      3: 'Confidential',
    },
    // -----------------------------------------
    // FILE PATHS
    document_paths: {
      general_attachments: '/attachments/general',
    },

    // -------------------------------------------
    // USER
    access: {
      0: 'Reader',
      1: 'Writer',
      2: 'Admin',
      3: 'Super Admin',
    },

    // -------------------------------------------
    // FEEDBACK
    feedback_types: {
      1: 'Bug',
      2: 'General Feedback',
    },
    feedback_status: {
      1: 'Open',
      2: 'On Hold',
      3: 'Closed (Complete)',
      4: 'Closed (Duplicate)',
      5: 'Closed (Feedback Only)',
    },
    // -----------------------------------------

    converter(items) {
      if (Array.isArray(items)) {
        return items
      }

      return Object.entries(items).map(([k, v]) => ({
        value: parseInt(k) || k,
        text: v,
      }))
    },
  })
}
